proto/outbound.proto

```diff
@@ -125,6 +125,7 @@ message HttpRoute {
       io.linkerd.proxy.http_route.HttpFailureInjector failure_injector = 1;
       io.linkerd.proxy.http_route.RequestHeaderModifier request_header_modifier = 2;
       io.linkerd.proxy.http_route.RequestRedirect redirect = 3;
+      io.linkerd.proxy.http_route.RequestHeaderModifier response_header_modifier = 4;
```

Comment:
Is it really appropriate to use the _RequestHeaderModifier_ type here when handling response headers? Are we absolutely sure these types will remain the same? What is the cost of introducing a distinct type?